core/templates/domain/objects/units-object.model.ts

```diff
@@ -0,0 +1,253 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of Units domain objects.
+ */
+
+
+import { createUnit, unit } from 'mathjs';
+import { ObjectsDomainConstants } from 'domain/objects/objects-domain.constants';
+import { Unit } from 'interactions/answer-defs';
+
+interface UnitsBackendDict {
+  units: Unit[];
+}
+
+interface UnitsDict {
+  [unit: string]: number;
+}
+
+type CurrencyUnitsKeys = (
+  keyof typeof ObjectsDomainConstants.CURRENCY_UNITS)[];
+
+export class Units {
+  units: Unit[];
+  constructor(unitsList: Unit[]) {
+    this.units = unitsList;
+  }
+
+  toDict(): UnitsBackendDict {
+    return {
+      units: this.units
+    };
+  }
+
+  toString(): string {
+    var unit = '';
+    for (var i = 0; i < this.units.length; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < this.units.length; i++) {
```
Prefer using `let` instead of `var`, here and elsewhere